cloudinit/tests/test_conftest.py

```diff
@@ -16,3 +18,23 @@ def test_typeerrors_on_incorrect_usage(self):
     @pytest.mark.parametrize('disable_subp_usage', [False], indirect=True)
     def test_subp_usage_can_be_reenabled(self):
         util.subp(['whoami'])
+
+
+class TestDisableSubpUsageInTestSubclass(CiTestCase):
+    """Test that disable_subp_usage doesn't impact CiTestCase's subp logic."""
+
+    def test_using_subp_raises_assertion_error(self):
+        with pytest.raises(Exception):
+            util.subp(["some", "args"])
+
+    def test_typeerrors_on_incorrect_usage(self):
+        with pytest.raises(TypeError):
+            util.subp()
+
+    def test_subp_usage_can_be_reenabled(self):
+        _old_allowed_subp = self.allow_subp
+        self.allowed_subp = True
+        try:
+            util.subp(['whoami'])
```

Comment:
I may be reading this wrong.  But I think this adds a test dependency on a program named 'whoami'.
That seems arbitrary and contrary to the point of this "don't use subp" functionality, which was to *not* depend on lots of programs in unit tests.

We explicitly test subp using BASH (tests/unittests/stest_util.py).

So it'd make sense to me to just use that same dependency here.

    try:
        util.subp([BASH, '-c', 'true'])
    finally:
        ...

Reply:
> That seems arbitrary and contrary to the point of this "don't use subp" functionality, which was to not depend on lots of programs in unit tests.

Well, this is testing that we can disable that behaviour, so it's intentionally contrary to that, else it wouldn't be much of a test. :-)

> We explicitly test subp using BASH (tests/unittests/stest_util.py).

`whoami` is in coreutils, so I very much doubt it will be absent in any cases where `bash` is available, but I'll change this for consistency.

Reply:
the motivation for not executing commands was really a bsd as much as some other  linux.